src/identity/components/GlobalHeader/GlobalHeaderDropdown/CreateOrganization/CreateOrganizationContext.tsx

```diff
@@ -0,0 +1,195 @@
+// Libraries
+import React, {
+  FC,
+  createContext,
+  ReactChild,
+  useState,
+  useEffect,
+  useCallback,
+} from 'react'
+
+// Components
+import {Cluster} from 'src/client/unityRoutes'
+
+// Types
+import {RemoteDataState} from 'src/types'
+
+// Utils
+import {createNewOrg, fetchClusterList} from 'src/identity/apis/org'
+import {useSelector} from 'react-redux'
+import {selectQuartzOrgs} from 'src/identity/selectors'
+
+interface Props {
+  children: ReactChild
+}
+
+export type ProviderID = 'AWS' | 'GCP' | 'Azure'
+export enum CreateOrgError {
+  NameConflict = 'name-conflict',
+  Unauthorized = 'unauthorized',
+  ServerError = 'server-error',
+  ClustersFetchError = 'clusters-fetch-error',
+  GenericError = 'generic-error',
+}
+
+export interface CreateOrgContextType {
+  changeCreateOrgLoadingStatus: (_: RemoteDataState) => void
+  changeCurrentProvider: (_: ProviderID) => void
+  changeCurrentRegion: (_: string) => void
+  changeOrgName: (_: string) => void
+  clusters: ClusterMap
+  createOrg: () => Promise<void>
+  createOrgLoadingStatus: RemoteDataState
+  currentRegion: string
+  currentProvider: ProviderID
+  error: CreateOrgError
+  orgName: string
+  setError: (_: CreateOrgError) => void
+}
+
+export const DEFAULT_CREATE_ORG_CONTEXT: CreateOrgContextType = {
+  changeCreateOrgLoadingStatus: (_: RemoteDataState) => null,
+  changeCurrentProvider: (_: ProviderID) => null,
+  changeCurrentRegion: (_: string) => null,
+  changeOrgName: (_: string) => null,
+  clusters: {},
+  createOrg: () => null,
+  createOrgLoadingStatus: RemoteDataState.NotStarted,
+  currentRegion: '',
+  currentProvider: null,
+  error: null,
+  orgName: '',
+  setError: (_: CreateOrgError) => null,
+}
+
+export const CreateOrgContext = createContext<CreateOrgContextType>(
+  DEFAULT_CREATE_ORG_CONTEXT
+)
+
+interface ClusterMap {
+  [providerId: string]: Cluster[]
+}
+
+export const CreateOrgProvider: FC<Props> = ({children}) => {
+  const quartzOrganizations = useSelector(selectQuartzOrgs)
```

Comment:
Instead of using this selector, you can use the more specific selector `selectQuartzOrgsContents`, which doesn't include the `.status` property.

Separately - I think we might need a `useEffect` here that fetches the list of organizations IF it's not currently loaded. In the future, we may be dynamically dispatching the thunk that fetches the user's organizations only after they click the 'switch' button in global header, in which case it'll be possible to get to this modal before the full org list has been fetched for the first time.

Reply:
nice!